changes/2024-TBD_cache-across-hierarchy-keyrings/change.md

```diff
@@ -210,29 +414,25 @@ it can be served without any KMS or DDB requests.
 If not, the recreated Hierarchy Keyring
 is used to refresh it.
 
-### Potential Helper Function: Hierarchy Keyring For KMS Relationship
-
-To push MPL Consumers in the correct direction,
-AWS Crypto Tools could write a helper method,
-that creates KeyStores with KMS Clients
-of a particular KMS Relationship,
-and sets the KeyStore's ID deterministically
-on the properties of the KMS Relationship.
-
-## A Looser Cache Entries (Branch Keys) Restriction
-
-Some MPL Consumers MAY not need such 
-## Out of Scope
+## A2: Sub-requirements
 
-TODO
+- We must update the cache identifier for [Hierarchy Keyring Branch Key Materials (Decryption) cache identifier](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/aws-kms/aws-kms-hierarchical-keyring.md#decryption-materials) to have a branch-key-digest=SHA512(branch-key-id) appended, instead of a branch-key-id, just like in the [Branch Key Materials (Encryption) cache identifier](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/aws-kms/aws-kms-hierarchical-keyring.md#encryption-materials) [[ref: Hierarchy Keyring Cache Identifier formula]](#hierarchy-keyring-cache-identifier-formula).
+- Minor update to the spec: The branch-key-version (which is the UUID) in the [Hierarchy Keyring Branch Key Materials (Decryption) cache identifier](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/aws-kms/aws-kms-hierarchical-keyring.md#decryption-materials) is currently mentioned to be of size 36 bytes. However, the UUID is actually only 16 bytes of randomness. We get 36 bytes when we convert the 16 bytes of randomness in UUID to UTF8 encoding. Therefore, the spec should mention that the branch-key-version has size 16 bytes [[ref: Hierarchy Keyring Cache Identifier formula]](#hierarchy-keyring-cache-identifier-formula).
```

Comment:
Suggestion: I know I used the term "entropy" and  "randomness", but technically, there is UUIDs are not a secure source of randomness.
 
```suggestion
- Minor update to the spec: The branch-key-version (which is the UUID) in the [Hierarchy Keyring Branch Key Materials (Decryption) cache identifier](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/aws-kms/aws-kms-hierarchical-keyring.md#decryption-materials) is currently mentioned to be of size 36 bytes. However, the UUID is actually only 16 bytes. We get 36 bytes when we convert the 16 bytes in UUID to a UTF8 encoding. Therefore, the spec should mention that the branch-key-version has size 16 bytes [[ref: Hierarchy Keyring Cache Identifier formula]](#hierarchy-keyring-cache-identifier-formula).
```

Reply:
Okay, thanks for pointing this out!